### PR TITLE
Expose protocols classes

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

Expose protocol classes (`Websocket`, `Http` and `SocketIO`) in the root module.  
Also remove the `networkWrapper`.

Documentation PR: https://github.com/kuzzleio/documentation-V2/pull/180
### How should this be manually tested?

Run this snippet and the unit tests:
```
const {
  Kuzzle,
  Websocket
} = require('./index');

const
  websocketProtocol = new Websocket({ host: 'localhost', port: 7512 }),
  kuzzle = new Kuzzle(websocketProtocol);

(async () => {
  try {
    await kuzzle.connect();

    console.log(await kuzzle.server.now());
  } catch (error) {
    console.log(error);
  } finally {
    kuzzle.disconnect();
  }
})()
```

### Other changes

 - Rename `KuzzleAbstractNetwork` to `KuzzleAbstractProtocol`
 - Rename network occurrences by protocol

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,9 +99,9 @@ import { Kuzzle } from 'kuzzle-sdk/dist/kuzzle.js'
 ```
 
 ## Protocols used
-Actually, the SDK support 3 protocols: `http`, `websocket` et `socketio`.  
+Actually, the SDK support 3 protocols: `Http`, `Websocket` and `SocketIO`.
```
Comment (scottinet) [inline README.md]: False friend:

```suggestion
Currently, the SDK support 3 protocols: `Http`, `Websocket` and `SocketIO`.  
```

File: README.md [hunk 2]
```diff
@@ -92,9 +99,9 @@ import { Kuzzle } from 'kuzzle-sdk/dist/kuzzle.js'
 ```
 
 ## Protocols used
-Actually, the SDK support 3 protocols: `http`, `websocket` et `socketio`.  
+Actually, the SDK support 3 protocols: `Http`, `Websocket` and `SocketIO`.  
 
-Websocket and Socket.IO protocols implement the whole Kuzzle API, while HTTP protocol does not implement realtime features (rooms and subscriptions).  
+Websocket and Socket.IO protocols implement the whole Kuzzle API, while **HTTP protocol does not implement realtime features** (rooms and subscriptions).
```
Comment (scottinet) [inline README.md]:
```suggestion
WebSocket and Socket.IO protocols implement the whole Kuzzle API, while the **HTTP protocol does not implement realtime features** (rooms and subscriptions).  
```

File: src/Kuzzle.js [hunk 3]
```diff
@@ -29,28 +28,23 @@ const
...
-    if (typeof network === 'string') {
-      return new Kuzzle(networkWrapper(network, options), options);
-    }
-
-    // custom protocol: check the existence of required methods
+    // check the existence of required methods
     for (const method of ['addListener', 'isReady', 'query']) {
-      if (typeof network[method] !== 'function') {
-        throw new Error(`Network instance must implement a "${method}" method`);
+      if (typeof protocol[method] !== 'function') {
```
Comment (scottinet) [inline src/Kuzzle.js]: Shouldn't we check that the provided protocol is an instance of `KuzzleAbstractProtocol` instead?

**Comment by ScreamZ** [comment]
Awesome good job @Aschen

**Pull request merged**